bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -261,7 +261,7 @@ crate::internal_macros::define_extension_trait! {
         /// Returns the minimum value an output with this script should have in order to be
         /// broadcastable on today’s Bitcoin network.
         #[deprecated(since = "0.32.0", note = "use `minimal_non_dust` etc. instead")]
-        fn dust_value(&self) -> Amount { self.minimal_non_dust() }
+        fn dust_value(&self) -> Option<Amount> { self.minimal_non_dust() }
```

Comment:
Didn't we agree that these are ridiculously unlikely and not worth it? I can't remember.

Reply:
It can only happen if something is attempting to produce a script that is 175000000 times longer than the block size. To my knowledge, everyone just uses one of the standard block templates, so it's valid by construction.

Reply:
Doing it like this makes the next change (making `from_sat` fallible) way easier. Last time I did not do that and it turned out too messy. So this is way cleaner. We can always re-visit if we don't want the `Option` and prefer to panic.

Reply:
How does it make it *way* easier if you can simply use `from_sat(x).unwrap()`?

Anyway, don't take this as blocking, just curiosity.

Reply:
Its easier because one doesn't have to reason about what values cause a panic - which I did last time and still got wrong.

> Anyway, don't take this as blocking, just curiosity.

Cool, thanks.

Reply:
Oh, I see. OK.

Reply:
Opened https://github.com/rust-bitcoin/rust-bitcoin/issues/4221

We should address this before release, but fine to leave it open during refactors.

Reply:
Yeah a downside to this change is it is disruptive.